src/preamble.js

```diff
@@ -1032,12 +1036,20 @@ function createWasm(env) {
     assert(Module === trueModule, 'the Module object should not be replaced during async compilation - perhaps the order of HTML elements is wrong?');
     trueModule = null;
 #endif
+#if LOAD_SOURCE_MAP
+    getSourceMapPromise().then(function (sourceMap) {
```

Comment:
I'm not sure `receiveInstantiatedSource` is the right place for this. This is basically just something that must be done before the program starts, I believe?

Doing it at the end of `instantiateAsync` seems more logical, basically just chaining it at the end of that promise?

We should also error if async compilation is off, since this can only work that way.